### PR TITLE
Document disabled execution optimizations for invalid work

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -24,6 +24,13 @@ For Java, Groovy, Kotlin and Android compatibility, see the [full compatibility
 
 <!-- Do not add breaking changes or deprecations here! Add them to the upgrade guide instead. --> 
 
+## Build reliability improvements
+
+Gradle employs a number of optimizations to ensure that builds are executed as fast as possible.
+These optimizations rely on the inputs and outputs of tasks to be well-defined.
+If a task is found to be invalid, Gradle will now execute it without the benefit of parallel execution, up-to-date checks and the build cache.
```
Comment (wolfs) [inline subprojects/docs/src/docs/release/notes.md:31]:
```suggestion
Gradle already applies some validation to tasks to check whether they are well defined.
If a task is found to be invalid, Gradle will now execute it without the benefit of parallel execution, up-to-date checks and the build cache.
```

It is unclear to me what `invalid` means from that sentence. I suggest adding a sentence that Gradle already does some validation.

File: subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc [hunk 2]
```diff
@@ -766,6 +766,19 @@ It tries to identify problems where e.g. annotations are used on incompatible ty
 Any getter not annotated with an input/output annotation is also flagged.
 These problems are then turned into deprecation warnings when the task is executed.
 
+[NOTE]
+====
+Tasks that fail validation are executed without any optimizations.
+Specifically, they never be:
```
Comment (wolfs) [inline subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc:772]:
```suggestion
Specifically, they never can be:
```

File: subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc [hunk 3]
```diff
@@ -766,6 +766,19 @@ It tries to identify problems where e.g. annotations are used on incompatible ty
...
+
+* <<##sec:up_to_date_checks,up-to-date>>,
+* loaded from or stored in the <<build_cache.adoc#build_cache,build cache>>,
+* executed in parallel with other tasks, even if <<multi_project_configuration_and_execution.adoc#sec:parallel_execution,parallel execution>> is enabled,
+* executed incrementally.
+
+The in-memory representation of the file system state (aka _Virtual File System)_ is also invalidated before an invalid task is executed.
```
Comment (wolfs) [inline subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc:779]: Should we add a link to the file system watching documentation here?

**Review by wolfs — COMMENTED** [review]
I left some comments.

File: subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc (hunk 3 above)
Comment (lptr) [reply subprojects/docs/src/docs/userguide/authoring-builds/more_about_tasks.adoc:779]: Added.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -24,6 +24,14 @@ For Java, Groovy, Kotlin and Android compatibility, see the [full compatibility
 
 <!-- Do not add breaking changes or deprecations here! Add them to the upgrade guide instead. --> 
 
+## Build reliability improvements
+
+Gradle employs a number of optimizations to ensure that builds are executed as fast as possible.
+These optimizations rely on the inputs and outputs of tasks to be well-defined.
+Gradle already applies some validation to tasks to check whether they are well defined.
```
Comment (wolfs) [inline subprojects/docs/src/docs/release/notes.md:31]:
```suggestion
Gradle already applies some validation to tasks to check whether they are well-defined.
```

Looks like I missed something in my suggestion.

**Comment by lptr** [comment]
@bot-gradle test QFL please.

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**